exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +83,531 @@ type BTCMarkets struct {
+	}
+	if clientOrderID != "" {
+		req["clientOrderID"] = clientOrderID
 	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsOrders, req, &resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetOrders returns current order information on the exchange
+func (b *BTCMarkets) GetOrders(marketID string, before, after, limit int64, status string) ([]OrderData, error) {
+	var resp []OrderData
+	params := url.Values{}
+	if marketID != "" {
+		params.Set("marketId", marketID)
+	}
+	if before >= 0 {
+		params.Set("before", strconv.FormatInt(before, 10))
+	}
+	if after >= 0 {
+		params.Set("after", strconv.FormatInt(after, 10))
+	}
+	if limit >= 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if status != "" {
+		params.Set("status", status)
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsOrders, params), nil, &resp)
+}
 
-		for j := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[j].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Volume /= common.SatoshisPerBTC
+// CancelAllOpenOrdersByPairs cancels all open orders unless pairs are specified
+func (b *BTCMarkets) CancelAllOpenOrdersByPairs(marketIDs []string) ([]CancelOrderResp, error) {
+	var resp []CancelOrderResp
+	req := make(map[string]interface{})
+	if len(marketIDs) > 0 {
+		var strTemp string
+		for x := range marketIDs {
+			strTemp += fmt.Sprintf("marketId=%s&", marketIDs[x])
 		}
+		req["marketId"] = strTemp[:len(strTemp)-1]
 	}
-
-	return resp.Orders, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders, req, &resp)
 }
 
-// GetOrderDetail returns order information an a specific order
-// orderID - example "1337"
-func (b *BTCMarkets) GetOrderDetail(orderID []int64) ([]Order, error) {
-	type OrderDetail struct {
-		OrderIDs []int64 `json:"orderIds"`
-	}
-	orders := OrderDetail{}
-	orders.OrderIDs = append(orders.OrderIDs, orderID...)
+// FetchOrder finds order based on the provided id
+func (b *BTCMarkets) FetchOrder(id string) (OrderData, error) {
+	var resp OrderData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	resp := Response{}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodPost, btcMarketsOrderDetail, orders, &resp)
-	if err != nil {
-		return nil, err
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals(before, after, limit int64) ([]TransferData, error) {
+	var resp []TransferData
+	params := url.Values{}
+	if before >= 0 {
```

Comment:
Please add a check to ensure that only before or after can be set and before should be > 0